state-chain/runtime/src/lib.rs

```diff
@@ -1262,6 +1262,23 @@ impl_runtime_apis! {
 		fn cf_free_balances(account_id: AccountId) -> Result<AssetMap<AssetAmount>, DispatchErrorWithMessage> {
 			LiquidityProvider::free_balances(&account_id).map_err(Into::into)
 		}
+		fn cf_lp_total_balances(account_id: AccountId) -> Result<AssetMap<AssetAmount>, DispatchErrorWithMessage> {
+			let free_balances = LiquidityProvider::free_balances(&account_id).map_err(Into::<DispatchErrorWithMessage>::into)?;
+			let open_order_balances = LiquidityPools::open_order_balances(&account_id);
+			let boost_pools_balances = {
+				let mut result = EthereumIngressEgress::boost_pool_account_balances(&account_id);
+				result.append( &mut PolkadotIngressEgress::boost_pool_account_balances(&account_id));
+				result.append( &mut BitcoinIngressEgress::boost_pool_account_balances(&account_id));
+				result.append( &mut ArbitrumIngressEgress::boost_pool_account_balances(&account_id));
+				result.append( &mut SolanaIngressEgress::boost_pool_account_balances(&account_id));
+				let mut map = AssetMap::from_fn(|_| 0);
+				for (asset, amount) in result {
+					*map.index_mut(asset) = amount;
```

Comment:
FYI I think this:

```
map.index_mut(asset) = x;
```

is equivalent to this:

```
map[asset] = x;
```

It can probably also be written bit more idiomatically, like collecting the results into a map and using that in AssetMap::from_fn. 

Not important though.